tests/drivers/grtc/grtc_clk_output/testcase.yaml

```diff
@@ -0,0 +1,61 @@
+common:
+  tags:
+    - drivers
+    - grtc
+    - ci_tests_zephyr_drivers_i2s
+  depends_on:
+    - gpio
+  harness: ztest
+
+tests:
+  drivers.grtc.grtc_clk_out.slow:
+    harness_config:
+      fixture: gpio_loopback
+    platform_allow:
+      - nrf54l15dk/nrf54l15/cpuapp
+      - nrf54h20dk/nrf54h20/cpuapp
+      - nrf54lm20apdk/nrf54lm20a/cpuapp
+      - nrf54lm20apdk@0.2.0/nrf54lm20a/cpuapp
+    integration_platforms:
+      - nrf54h20dk/nrf54h20/cpuapp
+
+  drivers.grtc.grtc_clk_out.fast.h20:
+    harness_config:
+      fixture: gpio_loopback
+    platform_allow:
+      - nrf54h20dk/nrf54h20/cpuapp
+    integration_platforms:
+      - nrf54h20dk/nrf54h20/cpuapp
+    extra_args:
+      - DTC_OVERLAY_FILE="boards/nrf54h20dk_nrf54h20_cpuapp_fast.overlay"
+    extra_configs:
+      - CONFIG_TEST_GRTC_FREQUENCY=50000
+      - CONFIG_TEST_GRTC_TOLERANCE=500
+
+  drivers.grtc.grtc_clk_out.fast.l15:
+    harness_config:
+      fixture: grtc_loopback
+    platform_allow:
+      - nrf54l15dk/nrf54l15/cpuapp
+    integration_platforms:
+      - nrf54l15dk/nrf54l15/cpuapp
+    extra_args:
+      - DTC_OVERLAY_FILE="boards/nrf54l15dk_nrf54l15_cpuapp_fast.overlay"
+    extra_configs:
+      - CONFIG_TEST_GRTC_FREQUENCY=40000
+      - CONFIG_TEST_GRTC_TOLERANCE=400
+
+  drivers.grtc.grtc_clk_out.fast.lm20:
+    harness_config:
+      fixture: gpio_loopback
+    platform_allow:
+      - nrf54lm20apdk/nrf54lm20a/cpuapp
+      - nrf54lm20apdk@0.2.0/nrf54lm20a/cpuapp
+    integration_platforms:
+      - nrf54lm20apdk/nrf54lm20a/cpuapp
+      - nrf54lm20apdk@0.2.0/nrf54lm20a/cpuapp
+    extra_args:
+      - DTC_OVERLAY_FILE="boards/nrf54lm20apdk_nrf54lm20a_cpuapp_fast.overlay"
+    extra_configs:
+      - CONFIG_TEST_GRTC_FREQUENCY=32000
+      - CONFIG_TEST_GRTC_TOLERANCE=320
```

Comment:
these should all be just one, use a fule suffix of "fast" then the overlay will be included, then you can have boards/<board>_fast.conf too which specifies the frequency and tolerance values